### PR TITLE
CMake project renaming, build improvements, and cleanups

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

/kind cleanup
/kind feature
/kind documentation

**Any specific area of the project related to this PR?**

/area build
/area libscap
/area libsinsp

**What this PR does / why we need it**:

This PR follows up the [OSS Libraries Contribution Plan](https://github.com/falcosecurity/falco/blob/master/proposals/2021019-libraries-donation.md) by introducing the following changes:

- rename the cmake project name to `falcosecurity-libs`
- make the driver name (and some other parameters) configurable via cmake options
- make [libscap.patch](https://github.com/falcosecurity/falco/blob/master/cmake/modules/falcosecurity-libs-repo/patch/libscap.patch) no longer needed (the patch was required by Falco due to a different naming convention) 
- cleanup some old references in the comments

The changes above mainly focus on the build system. Other naming adjustments will be addressed by following PRs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

~~Here you can find the testing branch in Falco: https://github.com/falcosecurity/falco/tree/build/apply-libs-pr-41
N.B.: that testing branch is rebased on top of https://github.com/falcosecurity/falco/pull/1671 to address an incompatible change introduced by https://github.com/falcosecurity/libs/commit/f7029e2522cc4c81841817abeeeaa515ed944b6c#diff-633539375239ac63e4a9e3358f0852d437b22e9c9f2c96de5e04e346b0b4784fL2074~~

TODOs:
- [x] rebase
- [x] use `scap` as default driver name
- [x] use `falcosecurity-libs` as cmake project name

**Does this PR introduce a user-facing change?**:

```release-note
build(userspace/libscap): make probe name configurable, default to `scap`
build: rename CMake project name to `falcosecurity-libs`
```

### CODE REVIEW

File: userspace/libscap/scap.c
Comment (leogr) [inline userspace/libscap/scap.c:118]: reminder to myself: could the missing `_probe` prefix break something?

**Review by gnosek — COMMENTED** [review]
Honestly, I'd prefer a top-level PRODUCT_NAME (or something) variable that all the variants get generated from. It feels weird to tie fs paths and env var names to the name of a kernel module (that happens to be the string we want)

Do you have a companion sysdig PR?

File: userspace/libscap/CMakeLists.txt
Comment (leogr) [reply userspace/libscap/CMakeLists.txt:56]: Actually, `HOST_ROOT` (without prefix) is the default value we want (it's used by Falco) when `PROBE_ENV_VAR_NAME` is not defined.

In general, I wanted to let any libs' consumer free to set whatever they want since those values can be significantly different in other contexts.
For this reason, I avoided introducing a top-level PRODUCT_NAME  variable

File: CMakeLists.txt
Comment (leogr) [reply CMakeLists.txt:35]: I can agree, but what else? :smile_cat:

File: userspace/libscap/CMakeLists.txt
Comment (gnosek) [reply userspace/libscap/CMakeLists.txt:56]: But you still want to set `SCAP_HOST_ROOT_ENV_VAR_NAME`, not `PROBE_ENV_VAR_NAME`, right?

Right now `PRODUCT_NAME` exists and is called `PROBE_NAME` (or maybe `PROBE_DEVICE_NAME`) whether you want it or not ;)

I'm not saying you should hardcode the policy to generate all these variables from PRODUCT_NAME (or rather, hardcode whatever you wish, but leave a way to supply your own)

Comment (leogr) [reply userspace/libscap/CMakeLists.txt:56]: You're right! I didn't notice that. I will fix it. Thank you!

**Comment by leogr** [comment]
> Honestly, I'd prefer a top-level PRODUCT_NAME (or something) variable that all the variants get generated from. It feels weird to tie fs paths and env var names to the name of a kernel module (that happens to be the string we want)

I intended to make each name configurable (without imposing any prefix/suffix) and using the PROBE_NAME as a fallback to generate defaults when values are not provided.

The env var name is already configurable (except for my mistake, but I will fix it soon). 
I can do a similar thing for the fs paths too.

**Comment by gnosek** [comment]
Any chance you can add a cmake var for the module name (in /sys/modules)? Something like:

```
string(REPLACE "-" "_" PROBE_MODULE_NAME "${PROBE_NAME}")
```

(I'm pretty sure I got something wrong, ref: https://cmake.org/cmake/help/latest/command/string.html).

I know it doesn't change anything for falco (which doesn't use the -probe suffix) but it would let OSS Sysdig (and the commercial agent) keep working without risky changes to the prebuilt probes tooling

File: userspace/libscap/CMakeLists.txt
Comment (leogr) [reply userspace/libscap/CMakeLists.txt:38]: I have added some commits, and all params should be configurable now.
The full flexibility should allow consumers to configure the probe and scap params as before, so nothing should break.

**Review by ldegio — APPROVED** [review]

**Comment by leogr** [comment]
/hold
until 2nd approval

File: CMakeLists.txt
Comment (araujof) [reply CMakeLists.txt:35]: Perhaps something like `falco-libs` or `falcolibs`?

Comment (leogr) [reply CMakeLists.txt:35]: What about the more verbose `falcosecurity-libs` (since it has been already used around)?

**Review by FedeDP — APPROVED** [review]

**Review by mstemm — APPROVED** [review]

**Review by ldegio — APPROVED** [review]

**Pull request merged**
